irods/connection.py

```diff
@@ -437,13 +456,32 @@ def _login_gsi(self):
 
     def _login_pam(self):
 
-        time_to_live_in_seconds = 60
-
-        pam_password = PAM_PW_ESC_PATTERN.sub(lambda m: '\\'+m.group(1), self.account.password)
+        import irods.client_configuration as cfg
+        inline_password = (self.account.authentication_scheme == self.account._original_authentication_scheme)
+        # By default, let server determine the TTL.
+        time_to_live_in_hours = 0
+        new_pam_password = PAM_PW_ESC_PATTERN.sub(lambda m: '\\'+m.group(1), self.account.password)
+        if not inline_password:
+            # Login in using PAM password from .irodsA
+            try:
+                self._login_native()
+            except (ex.CAT_PASSWORD_EXPIRED, ex.CAT_INVALID_USER, ex.CAT_INVALID_AUTHENTICATION):
+                time_to_live_in_hours = cfg.legacy_auth.pam.time_to_live_in_hours # TODO: document setting.
```

Comment:
TODO

Reply:
Do I need to change something about the comment?

Reply:
We should either do the thing or attach an issue number, yes. Preferably just do it since it's still fresh in your mind. :)

Reply:
documented in #500